src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -373,31 +379,50 @@ UCS_PROFILE_FUNC_ALWAYS(ucs_status_t, uct_ib_mlx5_devx_reg_indirect_key,
     return UCS_OK;
 }
 
+static uint32_t uct_ib_mlx5_symmetric_rkey_offset(uct_ib_mlx5_md_t *md)
+{
+    /* The direct key and atomic key must have constant offset so that remote
+     * keys can eventually compare equal.
+     *
+     * So first half of the range is used for allocation of smallest mkey,
+     * other half will be used to create atomic key explicitly.
+     */
+    return md->super.mkey_by_name_reserve.size / 2;
```

Comment:
maybe set this field to /2 value when it's read from FW?

Reply:
fixed